apps/src/javalab/JavabuilderConnection.js

```diff
@@ -48,13 +53,46 @@ export default class JavabuilderConnection {
   }
 
   onOpen() {
-    this.onOutputMessage('Compiling...');
     this.miniApp?.onCompile?.();
   }
 
+  onStatusMessage(messageKey) {
+    let message;
+    let includeLineBreak = false;
+    switch (messageKey) {
+      case StatusMessageType.COMPILING:
+        message = javalabMsg.compiling();
+        break;
+      case StatusMessageType.COMPILATION_SUCCESSFUL:
+        message = javalabMsg.compilationSuccess();
+        break;
+      case StatusMessageType.RUNNING:
+        message = javalabMsg.running();
+        includeLineBreak = true;
+        break;
+      case StatusMessageType.GENERATING_RESULTS:
+        message = javalabMsg.generatingResults();
+        break;
+      case StatusMessageType.EXITED:
+        this.onExit();
+        break;
+      default:
+        break;
+    }
+    if (message) {
+      this.onOutputMessage(`${STATUS_MESSAGE_PREFIX} ${message}`);
+    }
+    if (includeLineBreak) {
+      getStore().dispatch(appendNewlineToConsoleLog());
+    }
```

Comment:
Not a huge deal, but it feels a little weird that we're using `onOutputMessage` to abstract away the redux action for updating console logs but then directly using redux to append a new line? Do you think it would be cleaner to keep it consistent? If we don't have a need to abstract it, then it might be better to just use redux for both, but then again if we want to keep this component redux-free, it may be worth abstracting out both the `appendNewlineToConsole` action and `setIsRunning` actions. All that being said that might be a refactor outside the scope of this PR, so this is totally fine, but just wanted to call it out.